rgw/v2/tests/s3_swift/test_object_lock.py

```diff
@@ -0,0 +1,186 @@
+""" test_object_lock.py - Test object lock configuration for bucket
+
+Usage: test_object_lock.py -c <input_yaml>
+
+<input_yaml>
+	test_object_lock.yaml
+
+Operation:
+    Create bucket with bucket lock enabled
+    Enable object lock configuration
+    Upload object to the bucket
+    Upload another object in same bucket
+    Verify new version is created for object
+    Verify deletion of object versions are disabled
+"""
+
+
+import os
+import sys
+
+sys.path.append(os.path.abspath(os.path.join(__file__, "../../../..")))
+import argparse
+import datetime
+import hashlib
+import json
+import logging
+import time
+import traceback
+
+import boto3
+import botocore.exceptions as boto3exception
+import v2.lib.resource_op as s3lib
+import v2.utils.utils as utils
+from v2.lib.exceptions import AccessDeniedObjectDeleted, RGWBaseException, TestExecError
+from v2.lib.manage_data import io_generator
+from v2.lib.resource_op import Config
+from v2.lib.rgw_config_opts import CephConfOp, ConfigOpts
+from v2.lib.s3.auth import Auth
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+from v2.tests.s3_swift import reusable
+from v2.tests.s3cmd.reusable import get_rgw_ip_and_port
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+from v2.utils.utils import RGWService
+
+log = logging.getLogger()
+TEST_DATA_PATH = None
+password = "32characterslongpassphraseneeded".encode("utf-8")
+encryption_key = hashlib.md5(password).hexdigest()
+
+
+def test_exec(config):
+
+    io_info_initialize = IOInfoInitialize()
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+    ceph_conf = CephConfOp()
+    rgw_service = RGWService()
+
+    # create user
+    all_users_info = s3lib.create_users(config.user_count)
+    for each_user in all_users_info:
+        # authentication
+        auth = Auth(each_user, ssl=config.ssl)
+        rgw_conn = auth.do_auth()
+        # create buckets with object lock configuration
+        if config.test_ops["create_bucket"] is True:
+            log.info(f"no of buckets to create: {config.bucket_count}")
+            for bc in range(config.bucket_count):
+                bucket_name_to_create = utils.gen_bucket_name_from_userid(
+                    each_user["user_id"], rand_no=bc
+                )
+                log.info(f"creating bucket with name: {bucket_name_to_create}")
+                rgw_ip_and_port = get_rgw_ip_and_port()
+                s3_conn_client = boto3.client(
+                    "s3",
+                    aws_access_key_id=each_user["access_key"],
+                    aws_secret_access_key=each_user["secret_key"],
+                    endpoint_url=f"http://{rgw_ip_and_port}",
+                )
```

Comment:
remove this once you use `do_auth_using_client`

Reply:
Done.